### ISSUE TITLE
Cannot change/initialized the shape of a collision polygon through code?

### ISSUE THREAD

**Issue by Nodragem** [godotengine/godot#7040]
Hello,

I am trying to do some prototyping with Godot Engine. So I'm trying to use Polygon2D and CollisionPolygon2D.

It seems that there is a lack of default shape for Polygon2D, so I needed to create a circular shape through code:
```
extends CollisionPolygon2D

const shape = Vector2Array()
var util = load("util.gd").new()
	
func _ready():
	for i in range(0,360,10):
		shape.append(util.pol2cart(deg2rad(i), 50)+get_pos())
	print(shape)
	set_polygon(shape)
```

Then, instead of using CollisionShape/Circle, I wanted to simply import the Polygon2D into a CollisionPolygon2D (as it would be nice thing to do later for more complex shape):
``` python
extends CollisionPolygon2D
func _ready():
        set_build_mode(0)
	set_polygon(get_node("/root/main_scene/Polygon2D").get_polygon())
```

That is not working very much. Note that Godot asked me to create a shape for the CollisionPolygon2D object in the Godot 2D Editor (that is the object selected in the 2D editor, in the screenshot below) ; then my script is expected to overwrite that shape. Here the results:
![2016-11-05 3](https://cloud.githubusercontent.com/assets/10520249/20029698/88b104fe-a34b-11e6-8748-ee0baa49bbdb.png)
It does look like that the Collision polygon has been overridden by my script (you can see the red circle on the game window, drawn by the debugger), though with an offset. 
However, when I move the red square with my keyboard (that have the square collision shape, shown in grey), the red circle is not making any collision (I can pass through it). Interestingly, if I go, with my square, near the polygon shape that I defined in the 2D Editor, I can experience collision with that polygon (I cannot pass through it).

Thus, it seems that there may be:
- a bug in the debugger,
- a problem with creating polygon by code for the CollisionPolygon2D object.

After that I try to simply copy paste the code that creates my circular polygon for my Polygon2D object into the CollisionPolygon2D object:

```
extends CollisionPolygon2D

var util = load("util.gd").new()
const shape = Vector2Array()

func _ready():
	for i in range(0,360,10):
		shape.append(util.pol2cart(deg2rad(i), 50)+get_pos())
	print(shape)
	set_polygon(shape)
```

Surprisingly, that does not work at all :/ when I run the game, the debugger shows no collision polygon for the object. However, if I move my red square, I can see that there is collision with the shape defined in the 2D editor.

I don't know if that is a bug or if it is me who do it wrong.
But if it is a bug, it may be quite problematic for when people want to make some prototyping with simple shape.

**Comment by eon-s** [comment]
Remember that Collision shapes are used as editor helpers on the scene tree, are not nodes, these just make shape resources.
[](https://godot.readthedocs.io/en/stable/classes/class_collisionpolygon2d.html)

**Comment by vnen** [comment]
It's not a bug, it's just the way it works now. CollisionShapes are just editor helpers and once the game is running the shapes are inside the CollisionObject itself (Area2D, KinematicBody2D , etc.). To edit them via code in-game, you need to use the `*_shape()` functions from the [CollisionObject2D](http://docs.godotengine.org/en/stable/classes/class_collisionobject2d.html#class-collisionobject2d).

**Comment by Noshyaar** [comment]
This is resolved in latest master 9e7cee2
